### PR TITLE
V2.1 WIP MVP

### PR DESCRIPTION
**Pull request by tm21cy**
Extension of #13, while also adding functionality for button handling. See commits for full details.

### CODE REVIEW

File: src/buttons/join.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { languages } from "../types/help-roles";
+
+module.exports = {
+  name: "join",
+  async execute(interaction: ButtonInteraction) {
+    let cid = interaction.customId;
+    const userID = cid.split("-")[1];
+    let opts: SelectMenuOptionBuilder[] = [];
+    for (let lang of languages) {
+      opts.push(new SelectMenuOptionBuilder().setLabel(lang).setValue(lang));
```
Comment (avasilic) [inline src/buttons/join.ts:18]: Use proper formatting.
```suggestion
      opts.push(new SelectMenuOptionBuilder()
         .setLabel(lang)
         .setValue(lang)
      );
```

File: src/buttons/join.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    let menu = new StringSelectMenuBuilder()
+      .addOptions(opts)
+      .setCustomId(`jroles-${cid}`)
+      .setPlaceholder("Select languages to add.")
+      .setMaxValues(opts.length);
+
+    let row = new ActionRowBuilder<StringSelectMenuBuilder>().addComponents(
+      menu
+    );
```
Comment (avasilic) [inline src/buttons/join.ts:28]: Use proper formatting.
```suggestion
    let row = new ActionRowBuilder<StringSelectMenuBuilder>()
      .addComponents(
         menu
      );
```

File: src/buttons/join.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    let embed =
+      interaction.message.embeds[0] ??
+      new EmbedBuilder().setTitle("Select Language Roles").setColor("Blurple");
```
Comment (avasilic) [inline src/buttons/join.ts]: Prefer the usage of custom colours. Use `Colors.Indigo`. Make sure you import from the colours enum and not Discord.js.

File: src/buttons/leave.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  ActionRowBuilder,
+  ButtonInteraction,
+  EmbedBuilder,
+  SelectMenuOptionBuilder,
+  StringSelectMenuBuilder,
+} from "discord.js";
+import { languages } from "../types/help-roles";
+
+module.exports = {
+  name: "join",
```
Comment (avasilic) [inline src/buttons/leave.ts]: Use the proper name. 
```suggestion
  name: "leave",
```

File: src/commands/helper-roles.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    let row = new ActionRowBuilder<ButtonBuilder>().addComponents(
+      joinBtn,
+      leaveBtn
+    );
+    let embed = new EmbedBuilder()
+      .setTitle("Helper Role Membership Panel")
+      .setDescription(
+        "Using the buttons below to open interfaces, join and leave helper roles by interacting with the select menu."
+      )
+      .setColor("Blurple");
```
Comment (avasilic) [inline src/commands/helper-roles.ts]: Prefer the usage of custom colours. Use `Colors.Indigo`. Make sure you import from the colours enum and not Discord.js.

File: src/events/interactionHandler.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        }
+        break;
+
+      case InteractionType.MessageComponent:
+        const prefix = interaction.customId.split("-")[0];
+        switch (interaction.componentType) {
+          case ComponentType.Button:
+            let btn = interaction.client.buttons.get(`${prefix}`);
+            if (!btn) return;
+            await btn.execute(interaction);
```
Comment (avasilic) [inline src/events/interactionHandler.ts]: add a `.catch`

**Review by avasilic — CHANGES_REQUESTED** [review]
Review comments.

**Comment by tm21cy** [comment]
Changes authored.

File: src/commands/helper-roles.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  AutocompleteInteraction,
+  ButtonBuilder,
+  ButtonStyle,
+  ChatInputCommandInteraction,
+  EmbedBuilder,
+  SlashCommandBuilder,
+} from "discord.js";
+import { languages, platforms } from "../types/help-roles";
+import Color from "../enums/colors";
+import Colors from "../enums/colors";
```
Comment (avasilic) [inline src/commands/helper-roles.ts]: Undoubtedly one of these is a mistake? Why 2 imports?

File: src/events/interactionHandler.ts [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        break;
+
+      case InteractionType.MessageComponent:
+        const prefix = interaction.customId.split("-")[0];
+        switch (interaction.componentType) {
+          case ComponentType.Button:
+            let btn = interaction.client.buttons.get(`${prefix}`);
+            if (!btn) return;
+            try {
+              await btn.execute(interaction);
```
Comment (avasilic) [inline src/events/interactionHandler.ts]: As with the interaction command handler add a catch to this literally in addition to the try ... catch.

**Review by avasilic — CHANGES_REQUESTED** [review]
See comments.

**Comment by tm21cy** [comment]
Re-reviewed.

**Review by avasilic — APPROVED** [review]

**Pull request merged**